cpp/src/Ice/Reference.cpp

```diff
@@ -1094,11 +1094,13 @@ IceInternal::RoutableReference::toProperty(const string& prefix) const
         _endpointSelection == EndpointSelectionType::Random ? "Random" : "Ordered";
     {
         ostringstream s;
+        s.imbue(locale::classic()); // Ensure we use the C locale for the number formatting.
```

Comment:
I think we should use `to_string` or maybe `to_chars` for these integer-to-string conversions. No need for an ostringstream.